src/main/scala/mesosphere/marathon/api/v2/Validation.scala

```diff
@@ -201,12 +202,21 @@ object Validation {
     )
   }
 
-  def configValueSet[T <: AnyRef](config: String*): Validator[T] = {
-    new Validator[T] {
-      override def apply(t: T): Result = {
-        if (config.forall(AllConf.suppliedOptionNames)) Success else Failure(Set(RuleViolation(t,
-          s"""You have to supply ${config.mkString(", ")} on the command line.""", None)))
-      }
+  def configValueSet[T <: AnyRef](config: String*): Validator[T] =
+    isTrue(s"""You have to supply ${config.mkString(", ")} on the command line.""") { _ =>
+      config.forall(AllConf.suppliedOptionNames)
+    }
+
+  def isTrue[T](constraint: String)(test: T => Boolean): Validator[T] = new Validator[T] {
+    import ViolationBuilder._
+    override def apply(value: T): Result = {
+      if (test(value)) Success else RuleViolation(value, constraint, None)
     }
   }
+
+  class RuleValidator[T](val test: T => Boolean, val constraint: String) extends Validator[T] {
```

Comment:
Sure, your `isTrue` method is identical in terms of creating a `Vlidator` instance with modified apply method. RuleValidator did the same thing. I did delete it and am squashing the whole thing right now.